src/CMakeLists.txt

```diff
@@ -47,6 +47,7 @@ configure_file(csympy_config.h.in csympy_config.h)
 include_directories(BEFORE ${csympy_BINARY_DIR}/src)
 
 add_library(csympy ${SRC})
+target_link_libraries(csympy teuchos ${LIBS})
```

Comment:
For a), I think if you didn't put `teuchos` into `target_link_libraries`, then you need to explicitly add `teuchos` for each linking, as we did in tests for example. So I think it changes what libraries cmake appends during the link stage of programs that use csympy. I agree that it changes nothing how the `csympy.a` itself is linked.